.github/workflows/deploy.yml

```diff
@@ -19,18 +19,16 @@ on:
         - staging
         - production
         default: 'integration'
-  workflow_run:
-    workflows: [CI]
-    types: [completed]
-    branches: [main]
+  release:
+    types: [released]
 
 jobs:
   build-and-publish-image:
-    if: github.event_name == 'workflow_dispatch' || github.event.workflow_run.conclusion == 'success'
+    if: github.event_name == 'workflow_dispatch' || startsWith(github.ref_name, 'v') 
```

Comment:
This is quite a loose pattern - would this also match branches with names like `view-components`? Could that cause any confusion?

Reply:
This workflow is triggered by a release event or manually (workflow_dispatch event). If it's a "workflow_dispatch" event,  the first part of the "if" statement will always be true `github.event_name == 'workflow_dispatch'. So the second part of the "if" statement will only be reached on a release event. For release events `github.ref_name` will always by a reference to a git tag (never a branch name).

It's possible that someone manually creates a release with v prefix, but realistically manually release are rare on GOV.UK for our deployable apps.

Reply:
For reference, the docs are here for the `github` context object:
https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Whilst the docs describe `github.ref_name` as
> The short ref name of the branch or tag that triggered the workflow run. 

For release triggers, it's always [a tag that's triggered the run](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release).